c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,63 @@
-﻿using Newtonsoft.Json;
+﻿using JsonDiffPatchDotNet;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 using System;
 using System.Collections;
 using System.Collections.Generic;
 
 namespace c_sharp_json_diff
 {
-    public class Result {
-        public List<Diff> Diffs { get; set; } = new List<Diff>();
-
-    }
-    public class Diff
-    {
-        [JsonProperty("property")]
-        public string jsonProperty { get; set; } = "";
-        [JsonProperty("previous")]
-        public double previous { get; set; }
-        [JsonProperty("current")]
-        public double current { get; set; }
-        [JsonProperty("difference")]
-        public double difference { get; set; }
-    }
     public class JsonDiff
     {
-        public static Result GetDiff(IDictionary<string, double> first, IDictionary<string, double> second)
+        public static JObject NumericDiff(string left, string right)
         {
-           ICollection<String> firstKeys = first.Keys;
-            ICollection<String> secondKeys = second.Keys;
-            List<String> matches = new List<string>();
-            Result result = new Result();
-            foreach (string key in secondKeys)
-            {
-                if (firstKeys.Contains(key))
-                {
-                    matches.Add(key);
-                }
-            }
-            foreach(string key in matches)
+            string diff = JsonDiffBetweenTwo(left, right);
+            JObject diffJsonObject = JObject.Parse(diff);
+            foreach(KeyValuePair<string, JToken?> prop in diffJsonObject)
```

Comment:
To handle nested objects you probably need to split this method as follows:

1. First use the JsonDiffPatch library to turn the two objects into a single object
2. Then pass that single object to a recursive function

The recursive function could do as follows. I'm not sure of the exact mechanics but something like this:

1. Start with a for-loop like you have
2. If the ``JToken`` is a "leaf", i.e. no children, then the code will look like what you have
3. Else, recursively call the recursive function on that ``JToken``

The above assumed that the iteration through ``JToken``s is iterating through the top level children only of the ``JObject``